client/phishery_docx.py

```diff
@@ -91,6 +133,36 @@ def signal_send_precheck(self, _):
 		if target_url is None:
 			self.text_insert('The phishery target URL is invalid.\n')
 			return False
+
+		if not self.config['add_landing_pages']:
+			return True
+		options = {'campaign': self.application.config['campaign_id']}
+		results = rpc.graphql("""\
+			query getCampaignExport($campaign: String!) {
+				db {
+					campaign(id: $campaign) {
+						landingPages {
+							edges {
+								node {
+									id
+									campaignId
+									hostname
+									page
+								}
+							}
+						}
+					}
+				}
+			}""", options)
+		document_urls = target_url.split()
+		landing_pages = [(node['node']['hostname'], node['node']['page']) for node in results['db']['campaign']['landingPages']['edges']]
+		for document_url in document_urls:
+			parsed_url = urllib.parse.urlparse(document_url)
+			hostname = parsed_url.netloc
+			landing_page = parsed_url.path
+			landing_page.lstrip()
+			if (hostname, landing_page) not in landing_pages:
+				self.application.rpc('campaign/landing_page/new', self.application.config['campaign_id'], hostname, landing_page)
```

Comment:
Yeah... this method only modifies the database if the new landing page is not already in the database. You don't have to check yourself and can completely skip the graphql query, parsing the url etc. All you need to do is call this rpc method for each landing page with the campaign id and hostname. It checks if the landing page already exists server side.